pkg/grpc/actions/integrations/update_integration.go

```diff
@@ -0,0 +1,82 @@
+package integrations
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"github.com/superplanehq/superplane/pkg/authentication"
+	"github.com/superplanehq/superplane/pkg/crypto"
+	"github.com/superplanehq/superplane/pkg/grpc/actions"
+	"github.com/superplanehq/superplane/pkg/models"
+	pb "github.com/superplanehq/superplane/pkg/protos/integrations"
+	"github.com/superplanehq/superplane/pkg/registry"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func UpdateIntegration(
+	ctx context.Context,
+	encryptor crypto.Encryptor,
+	registry *registry.Registry,
+	domainType, domainID, idOrName string,
+	spec *pb.Integration,
+) (*pb.UpdateIntegrationResponse, error) {
+	_, userIsSet := authentication.GetUserIdFromMetadata(ctx)
+	if !userIsSet {
+		return nil, status.Error(codes.Unauthenticated, "user not authenticated")
+	}
+
+	if spec == nil || spec.Metadata == nil || spec.Metadata.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "integration name is required")
+	}
+
+	integration, err := findIntegration(domainType, uuid.MustParse(domainID), idOrName)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	updatedIntegration, err := buildIntegration(ctx, encryptor, registry, domainType, uuid.MustParse(domainID), spec)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	integration.Name = updatedIntegration.Name
+	integration.Type = updatedIntegration.Type
+	integration.URL = updatedIntegration.URL
+	integration.AuthType = updatedIntegration.AuthType
+	integration.Auth = updatedIntegration.Auth
+
+	err = integration.Update()
```

Comment:
What happens if an integration was already used to set up some webhooks into an account, and we update it to point to something else?